### PR TITLE
Standarize locale comparisons and sort icons by title in package

### LINKED ISSUE #7851: Configure region in locales comparisons
We are using `String.prototype.localeCompare` to sort icons in *scripts/lint/ourlint.js* and *.svglintrc.mjs*:

https://github.com/simple-icons/simple-icons/blob/d5444f699528a456ee2894751994b4eef7274235/scripts/lint/ourlint.js#L21-L26

https://github.com/simple-icons/simple-icons/blob/d5444f699528a456ee2894751994b4eef7274235/.svglintrc.mjs#L49

According to the [mdn web docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare), we are not specifying the locale used to compare, so the execution could be different depending on the implementation. Also, `String.prototype.localeCompare` creates a new [`Intl.Collator`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator) object, which can be avoided just creating the object before and using it to compare.

### Contributing

Yes

### PR DESCRIPTION
**Pull request by mondeja**
Resolves #7851 and contributes to an improvement over #7852

### CODE REVIEW

File: .svglintrc.mjs [hunk 1]
```diff
@@ -46,7 +47,7 @@ const sortObjectByKey = (obj) => {
 
 const sortObjectByValue = (obj) => {
   return Object.keys(obj)
-    .sort((a, b) => ('' + obj[a]).localeCompare(obj[b]))
+    .sort((a, b) => collator.compare(`${obj[a]}`, obj[b]))
```
Comment (LitoMore) [inline .svglintrc.mjs:50]:
```suggestion
    .sort((a, b) => collator.compare(obj[a], obj[b]))
```

Comment (mondeja) [reply .svglintrc.mjs]: It seems that is not needed, are always all strings.

**Comment by LitoMore** [comment]
What if two brands have different titles but the same slug?

**Comment by mondeja** [comment]
The next error will raise at build time:

<details>
  <summary>Error</summary>

```
$ npm run build

> simple-icons@7.12.0 build
> node scripts/build/package.js

/home/mondeja/files/code/simple-icons/node_modules/esbuild/lib/main.js:1624
  let error = new Error(`${text}${summary}`);
              ^

Error: Transform failed with 2 errors:
<stdin>:10789:13: ERROR: Multiple exports with the same name "siHandshake_protocol"
<stdin>:10789:13: ERROR: The symbol "siHandshake_protocol" has already been declared
    ... <TRACEBACK> ... {
  errors: [
    {
      detail: undefined,
      id: '',
      location: {
        column: 13,
        file: '<stdin>',
        length: 20,
        line: 10789,
        lineText: 'export const siHandshake_protocol={',
        namespace: '',
        suggestion: ''
      },
      notes: [
        {
          location: {
            column: 13,
            file: '<stdin>',
            length: 20,
            line: 10777,
            lineText: 'export const siHandshake_protocol={',
            namespace: '',
            suggestion: ''
          },
          text: 'The name "siHandshake_protocol" was originally exported here:'
        }
      ],
      pluginName: '',
      text: 'Multiple exports with the same name "siHandshake_protocol"'
    },
    {
      detail: undefined,
      id: '',
      location: {
        column: 13,
        file: '<stdin>',
        length: 20,
        line: 10789,
        lineText: 'export const siHandshake_protocol={',
        namespace: '',
        suggestion: ''
      },
      notes: [
        {
          location: {
            column: 13,
            file: '<stdin>',
            length: 20,
            line: 10777,
            lineText: 'export const siHandshake_protocol={',
            namespace: '',
            suggestion: ''
          },
          text: 'The symbol "siHandshake_protocol" was originally declared here:'
        }
      ],
      pluginName: '',
      text: 'The symbol "siHandshake_protocol" has already been declared'
    }
  ],
  warnings: []
}
```

</details>

**Review by LitoMore — APPROVED** [review]

**Pull request merged**